# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/ipc/RemoteVideoDecoder.h
# Commit: 57607f8b6d8e
# Full Hash: 57607f8b6d8e1619b5c488494fd57547944c47b1
# Author: Dan Glastonbury <dan.glastonbury@gmail.com>
# Date: 2020-10-21 09:51:28
# Regressor Bug: 1595994
# File Overlap Count: 1
# Description:
#   Bug 1595994 - P7: Add Supports messages to PRemoteDecoderManager. r=kamidphish,nika,ipc-reviewers
#   
#   Add a synchronous Supports message to the IPDL PRemoteDecoderManager protocol so
#   decoder modules can query for playback support in the actual process that will
#   attempt to do the decoding.
# ==============================================================================

diff -r 1932f2f34993 -r 57607f8b6d8e dom/media/ipc/RemoteVideoDecoder.h
--- a/dom/media/ipc/RemoteVideoDecoder.h	Tue Oct 20 23:26:25 2020 +0000
+++ b/dom/media/ipc/RemoteVideoDecoder.h	Tue Oct 20 23:26:27 2020 +0000
@@ -17,7 +17,20 @@
 
 namespace mozilla {
 
-class KnowsCompositorVideo;
+class KnowsCompositorVideo : public layers::KnowsCompositor {
+ public:
+  NS_INLINE_DECL_THREADSAFE_REFCOUNTING(KnowsCompositorVideo, override)
+
+  layers::TextureForwarder* GetTextureForwarder() override;
+  layers::LayersIPCActor* GetLayersIPCActor() override;
+
+  static already_AddRefed<KnowsCompositorVideo> TryCreateForIdentifier(
+      const layers::TextureFactoryIdentifier& aIdentifier);
+
+ private:
+  KnowsCompositorVideo() = default;
+  virtual ~KnowsCompositorVideo() = default;
+};
 
 using mozilla::ipc::IPCResult;
 